saleor/graphql/product/mutations/product/product_update.py

```diff
@@ -43,19 +50,27 @@ class Meta:
         support_meta_field = True
         support_private_meta_field = True
 
+    FIELDS_TO_TRACK = list(PRODUCT_UPDATE_FIELDS)
+
     @classmethod
     def get_instance(cls, info, **data):
         """Prefetch related fields that are needed to process the mutation."""
-        # If we are updating an instance and want to update its attributes,
-        # prefetch them.
         object_id = cls.get_object_id(**data)
-        if object_id and data.get("attributes"):
-            # Prefetches needed by ProductAttributeAssignmentMixin and
-            # associate_attribute_values_to_instance
-            qs = cls.Meta.model.objects.prefetch_related(
-                "product_type__product_attributes__values",
-                "product_type__attributeproduct",
-            )
+        if object_id:
+            prefetch: list[str] = []
+            if data["input"].get("attributes"):
+                # Prefetches needed by ProductAttributeAssignmentMixin and
+                # associate_attribute_values_to_instance
+                prefetch.extend(
+                    (
+                        "product_type__product_attributes__values",
+                        "product_type__attributeproduct",
+                    )
+                )
```

Comment:
On main branch we were skipping this part due to wrong logical expression. However, commenting this part results with **less** number of queries executed. Any idea what is wrong here?